content/guides/hot-module-replacement.md

```diff
@@ -18,125 +19,194 @@ related:
+    console.log('Accepting the updated printMe module!');
+    document.body.removeChild(element);
+    element = component(); // update binding now that new printMe module has been acepted
+    document.body.appendChild(element);
+  })
 }
-```
 
-Even if the underlying module containing this class is patched with new code, any existing instances of the class still have the old `log` method. Meaning if we changed what that method does, it wouldn't be reflected in those old instances unless we re-instantiate them somehow using `module.hot.accept`.
+let element = component();
 
-This is just one example, but there are many others that can easily trip people up. Luckily, there are a lot of loaders out there, some mentioned below, that will make using this process much easier.
+function component() {
+  var element = document.createElement('div');
+  var btn = document.createElement('button');
 
+  element.innerHTML = _.join(['Hello', 'webpack'], ' ');
 
-## HMR with Stylesheets
+  btn.innerHTML = 'Click me and check the console!';
+  btn.onclick = printMe;  // onclick event is bind to the original printMe function
 
-We can use the `style-loader` to achieve Hot Module Replacement with CSS. This loader uses `module.hot.accept` behind the scenes to patch `<style>` tags when CSS dependencies are updated. So, with the following webpack configuration...
+  element.appendChild(btn);
 
-``` js
-module.exports = {
-  // ...
-  module: {
-    rules: [
-      {
-        test: /\.css$/,
-        use: [ 'style-loader', 'css-loader' ]
-      }
-    ]
-  },
-  // ...
+  return element;
 }
+
+document.body.appendChild(element);
+```
+
+This is just one example, but there are many others that can easily trip people up. Luckily, there are a lot of loaders out there, some mentioned below, that will make using this process much easier.
+
+## HMR with Stylesheets
+
+Hot Module Replacement with CSS is actually fairly straightforward with the help of the `style-loader`. This loader uses `module.hot.accept` behind the scenes to patch `<style>` tags when CSS dependencies are updated.
+
+Now let's update the configuration file to make use of the loader.
+
+```diff
+  const path = require('path');
+  const HtmlWebpackPlugin = require('html-webpack-plugin');
+  const webpack = require('webpack');
+
+  module.exports = {
+    entry: {
+      app: './src/index.js',
+      print: './src/print.js'
+    },
+    devtool: 'inline-source-map',
+    devServer: {
+      contentBase: './dist',
+      hot: true
+    },
++   module: {
++     rules: [
++       {
++         test: /\.css$/,
++         use: ['style-loader', 'css-loader']
++       }
++     ]
++   },
+    plugins: [
+      new HtmlWebpackPlugin({
+        title: 'Hot Module Replacement'
+      }),
+      new webpack.HotModuleReplacementPlugin()
+    ],
+    output: {
+      filename: '[name].bundle.js',
+      path: path.resolve(__dirname, 'dist')
+    }
+  };
 ```
 
 hot loading stylesheets is a breeze...
```

Comment:
I would change this slightly to keep the tone a little more friendly, maybe:

`Hot loading stylesheets is as easy as importing them into a module:`